runtime/doc/lua-plugin.txt

```diff
@@ -0,0 +1,443 @@
+
+However, some benefits of |<Plug>| mappings over this are that you can
+
+- Enforce options like `expr = true`.
+- Expose functionality only for specific |map-modes|.
+- Expose different behavior for different |map-modes| with a single |<Plug>|
+  mapping, without adding impurity or complexity to the underlying Lua
+  implementation.
+
+NOTE: If you have a function that takes a large options table, creating lots
+      of |<Plug>| mappings to expose all of its uses could become
+      overwhelming. It may still be beneficial to create some for the most
+      common ones.
+
+------------------------------------------------------------------------------
+Example					      *lua-plugin-plug-mapping-example*
+
+In your plugin:
+>lua
+vim.keymap.set("n", "<Plug>(SayHello)", function() 
+    print("Hello from normal mode") 
+end, { noremap = true })
+
+vim.keymap.set("v", "<Plug>(SayHello)", function() 
+    print("Hello from visual mode") 
+end, { noremap = true })
+<
+In the user's config:
+>lua
+vim.keymap.set({"n", "v"}, "<leader>h", "<Plug>(SayHello)")
+<
+==============================================================================
+Initialization					    *lua-plugin-initialization*
+
+Newcomers to Lua plugin development will often put all initialization logic in
+a single `setup` function, which takes a table of options.
+If you do this, users will be forced to call this function in order to use
+your plugin, even if they are happy with the default configuration.
+
+Cleanly separated configuration and smart initalization allow your plugin to 
+work out of the box.
+
+NOTE: A well designed plugin has minimal impact on startup time.
+      See also |lua-plugin-lazy-loading|.
+
+Common approaches to a cleanly separated configuration are:
+
+- A Lua function, e.g. `setup(opts)` or `configure(opts)`, which only overrides the
+  default configuration and does not contain any initialization logic.
+- A Vimscript compatible table (e.g. in the |vim.g| or |vim.b| namespace) that your
+  plugin reads from and validates at initialization time.
+  See also |lua-vim-variables|.
+
+Typically, automatic initialization logic is done in a |plugin| or |ftplugin|
+script. See also |runtimepath|.
+
+==============================================================================
+Lazy loading					      *lua-plugin-lazy-loading*
+
+When it comes to initializing your plugin, assume your users may not be using
+a plugin manager that takes care of lazy loading for you.
+Making sure your plugin does not unnecessarily impact startup time is your
+responsibility. A plugin's functionality may evolve over time, potentially
+leading to breakage if users have to hack into the loading mechanisms.
+Furthermore, a plugin that implements its own lazy initialization properly will
+likely have less overhead than the mechanisms used by a plugin manager or user
+to load that plugin lazily.
+
+Think carefully about when which parts of your plugin need to be loaded.
```

Comment:
Annoying sentence. "Think smart and lots, don't be dumb"

Reply:
I guess I could just remove it? The relevant info is in the subsections that follow after it.

Reply:
I think so. I'm aware this is well into nitpick territory but whatever, you'll live

Reply:
I'm very happy for your feedback :pray:  :smile: